ls1intum/Artemis
# Add human readable breadcrumbs for all remaining management routes

### Checklist


### Motivation and Context

Not all routes were displaying human readable breadcrumbs yet.

### Description

This changes the approach to no longer require child routes to be set up. Instead, routes are registered in the navbar, translations added in a map and IDs resolved using the respective services and not resolvers. (See the new documentation and the inline code if you are interested in the details).

### Steps for Testing

Below I listed all routes that should now display proper crumbs. It should 
display translated strings and titles instead of IDs where possible (I noted the exceptions). All breadcrumbs should be clickable (and take you somewhere instead of erroring).

I have implemented workarounds for routes which do not correspond to our ideal layout. Those will be refactored in a follow-up PR (to not make this more complex than it already is).

```
Server Administration:
- Upcoming Exams
- User Management
  - Create
  - Single user view (displays user login instead of ID)
  - Single user edit (displays user login instead of ID)
- System Notifications
  - Create
  - Single notification view (displays ID as there is no title)
  - Single notification edit (displays ID as there is no title)
- Featurer toggles
- User statistics
- Metrics
- Health
- Configuration
- Audits
- Logs
- API

Course Management:
- Create new course
- Single course
  - Edit
  - Q&A
  - All three course groups
  - Learning goal management
    - Create
    - Edit
  - Assessment Dashboard
  - Instructor Dashboard
  - Course Scores

Lectures (part of course management):
- Create
- View
- Edit
- Attachments
- Units
  - Create: Text, Video, Attachment, Exercise (do not display an ID as there is not a single view outside of the container)
  - Edit: Text, Video, Attachment (do not display an ID as there is not a single view outside of the container)

Exercises (part of course management):
- For apollon diagram/drag and drop quiz:
  - View/Open, Create
- For all other exercise types:
  - Create, Edit, Scores, Participations
- Quiz exercises only: Preview, Sample Solution, Statistics, Edit, Re-evaluate
- For programming, modeling, text, and file upload exercises: View, Submissions and Assessments/Grading, Plagiarism Detection
- For text/modeling exercises: Creating example submissions and solutions (do not display IDs or new but just "Example Solution"/"Example Submissions Editor" instead)
- Exercise Assessment Dashboard for all exercises
- Exercise Instructor Dashboard for all exercises
- Assess exercises (all types)
- For programming exercises:
  - Code editor
  - Hints: View, Create, Edit

Exams (part of course management):
- Create
- For a single exam:
  - View/Id or checklist (displays exam title)
  - Edit
  - Student Exams
  - Assessment Dashboard
  - Students
  - Test runs:
    - Start (conduction) displays "Exam"
    - Assessment dashboard
    - Summary
  - Scores
  - Exercise groups:
    - Exercise buttons work the same as for course exercises
    - "Add" buttons don't have extra breadcrumbs (only display until "exercise groups")
```

The dropdowns in the navigation bar should also all still work.

### Test Coverage

app/shared/layouts/navbar/navbar.component.ts: `0%` -> `81.01%`
